### PR TITLE
Clean

### PR DESCRIPTION
**Pull request by tsiguenz**

### CODE REVIEW

File: backend/src/2fa/2fa.service.ts [hunk 1]
```diff
@@ -2,14 +2,11 @@ import { Injectable } from '@nestjs/common';
...
 import { PrismaService } from '../prisma/prisma.service';
-import { User } from '@prisma/client';
-import { Request } from 'express';
 
 @Injectable()
 export class TwoFaService {
   constructor(private prisma: PrismaService) {}
-  async createSecret(req: Request) {
-    const userId = req.user['id'];
+  async createSecret(userId: number) {
```
Comment (Crysicia) [inline backend/src/2fa/2fa.service.ts:9]: It's definitely cleaner, thanks!

File: backend/src/profile/profile.service.ts [hunk 2]
```diff
@@ -13,18 +12,20 @@ export class ProfileService {
     private jwt: JwtService,
     private users: UsersService
   ) {}
-  async getProfile(req: Request) {
+  async getProfile(userId: number) {
```
Comment (Crysicia) [inline backend/src/profile/profile.service.ts:15]: Same here, good job!

File: backend/src/profile/profile.service.ts [hunk 3]
```diff
@@ -13,18 +12,20 @@ export class ProfileService {
...
         where: {
-          nickname: nickname
+          id: userId
+        },
+        select: {
+          id: true,
+          nickname: true,
+          avatar: true,
+          createdAt: true,
+          twoFactorEnable: true
```
Comment (Crysicia) [inline backend/src/profile/profile.service.ts:26]: Yes! That's good!

File: backend/src/profile/profile.service.ts [hunk 4]
```diff
@@ -33,22 +34,20 @@ export class ProfileService {
     }
   }
 
-  async editProfile(dto: EditProfileDto, req: Request) {
+  async editProfile(dto: EditProfileDto, userId) {
```
Comment (Crysicia) [inline backend/src/profile/profile.service.ts:37]: ⚠️ Missing a type definition here

**Review by gmorange42 — APPROVED** [review]

**Pull request merged**